wordpress-mobile/WordPress-Android
# [Modal Layout Picker] Create Layout Picker sections and thumbnail UI using static data

**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2436

**Related PRs:**
- **Depends on**: https://github.com/wordpress-mobile/WordPress-Android/pull/12482
- Followed by: https://github.com/wordpress-mobile/WordPress-Android/pull/12573 (and other [Modal Layout Picker tasks](https://github.com/wordpress-mobile/gutenberg-mobile/issues?q=is%3Aissue+is%3Aopen+%5BModal+Layout+Picker%5D))

## Description

This PR implements the presentation of static layout data as part of the effort to rework the Starter Page Template Picker for new page.
In detail it implements the following:
+ Layouts grouped into sections based on their category and horizontally scrollable within their category rows. 
+ Preview image on each layout cell
+ Selecting a Layout:
    + Updates the border-color
    + Displays the checkmark
    + Changes the bottom display to have buttons for "Preview" and "Create Page"
    + Deselects any currently selected cells
+ Deselecting a Layout:
    + Reverts the border-color.
    + Hides the checkmark.
    + Revert the bottom Buttons to be "Create Blank Page"
+ Handles Dark Mode for the new UI items. 

## To test:

Layout Picker should show [when creating a new page from *My Site* or *Site Pages* the *Modal Layout Picker* appears](https://github.com/wordpress-mobile/WordPress-Android/pull/12482).

**Note**: For demo/testing purposes, the categories repeat the same layout 30 times because the static data only has one layout per category. 

### General Navigation
1. Category rows should be horizontally scrollable within their sections with a preview image
    - Pick a row and scroll horizontally. 
    - Expect the category to scroll to the end and back again
1. Scrolling down and back up on the page should retain the scroll position of the category row
    - Pick a row and scroll horizontally.
    - Scroll vertically.
    - Scroll back to your reference row and expect it to be in the same location
1. Scrolling down and back up on the page should retain the selected layout
    - Select a layout
    - Scroll vertically and/or horizontally.
    - Scroll back to your reference layout and expect it to be selected still
### Preview Images
1. Layouts that haven't loaded yet should display a preview placeholder.
1. Images should be cached for use offline
    - Open the Layout picker and allow some images to download
    - Leave the layout picker
    - Either go offline or monitor the network calls
    - Open the layout picker
    - Expect the images for the layout picker to be pulled from the cache
### Selecting a Layout
1. Selecting a layout should highlight the cell and add a checkmark
    - Open the layout picker and select a cell
1. Only one layout should be selected at any given time
    - Open the layout picker and select a cell
    - Select a different cell (either in the same row or a different row)
    - Expect the old cell to deSelect and the new one to be selected
1. Selecting a Layout should show the buttons "Preview" and "Create Page"
    - Open the layout picker and select a cell
    - *Note*: The "Preview" and "Create Page" actions will be addressed by [#2452](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2452) and [#2446](https://github.com/wordpress-mobile/gutenberg-mobile/issues/2446)
1. Deselecting a layout should revert the bottom bar to "Create Blank Page"
    - Open the layout picker and select a cell
    - Deselect the cell by tapping on it again
    - Expect the bottom buttons to revert to "Create Blank Page"

## Screenshots:
| Top scroll | Middle scroll - selection |
|---|---|
|![device-2020-08-07-134403](https://user-images.githubusercontent.com/304044/89638538-aad4b100-d8b4-11ea-8e54-cb9605797cfe.png)|![device-2020-08-07-134423](https://user-images.githubusercontent.com/304044/89638573-b922cd00-d8b4-11ea-86f8-c2d175fcc980.png)|

| Top scroll (Dark Mode) | Middle scroll - selection (Dark Mode) |
|---|---|
|![device-2020-08-07-134521](https://user-images.githubusercontent.com/304044/89639386-19fed500-d8b6-11ea-83dc-fcb7763dd9ee.png)|![device-2020-08-07-134541](https://user-images.githubusercontent.com/304044/89639395-1d925c00-d8b6-11ea-953b-2af6e71549f4.png)|

|Landscape - selection |
|---|
|![device-2020-08-07-134455](https://user-images.githubusercontent.com/304044/89639128-b2488a00-d8b5-11ea-97d5-8013cf01373d.png)|

| Image placeholder | Action |
|---|---|
|![device-2020-08-07-134051](https://user-images.githubusercontent.com/304044/89639511-59c5bc80-d8b6-11ea-965e-fcdc5cec08b8.png)|![w](https://user-images.githubusercontent.com/304044/89639659-a6a99300-d8b6-11ea-8955-84f890814108.gif)|

PR submission checklist:
